### ISSUE TITLE
%base Desk Clean-Up

### ISSUE THREAD

**Issue by sigilante** [urbit/urbit#7004]
The `%base` desk bears the marks of many layers of development.  Ideally, the `%base` desk would consist of exactly what is necessary to produce a working Urbit instance—`/sys` and Azimuth support.  We're a long way from that, and there doesn't seem to be much kelvin pressure yet to actually strip it down.

Instead, let's take a look at what's in there (particularly in `/gen` and `/ted`) and update, fix, and document (or remove) the components of `%base`.

```
gen
├── acme
│   └── domain-validation.hoon
├── agents.hoon
├── ames-flows.hoon
├── ames-keys.hoon
├── ames-timers.hoon
├── aqua
│   ├── breach.hoon
│   ├── dojo.hoon
│   ├── file.hoon
│   ├── init-azimuth.hoon
│   ├── init.hoon
│   ├── pill.hoon
│   ├── raw-event.hoon
│   ├── restore-fleet.hoon
│   ├── scry.hoon
│   ├── snap-fleet.hoon
│   └── spawn.hoon
├── aqua-export.hoon
├── azimuth
│   ├── listen.hoon
│   └── watch.hoon
├── azimuth-block.hoon
├── azimuth-rpc
│   └── disconnect.hoon
├── azimuth-sources.hoon
├── brass.hoon
├── cat.hoon
├── code.hoon
├── configure-hosting.hoon
├── cors-registry.hoon
├── dbug.hoon
├── deco.hoon
├── desk-jam.hoon
├── desk-requests.hoon
├── dns
│   ├── auto.hoon
│   └── request.hoon
├── dns-bind
│   └── authority.hoon
├── dns-collector
│   └── complete.hoon
├── dojo
│   ├── acl.hoon
│   ├── allow-remote-login.hoon
│   ├── revoke-remote-login.hoon
│   └── wipe.hoon
├── gall-nonces.hoon
├── hello.hoon
├── help.hoon
├── hood
│   ├── ames-cong.hoon
│   ├── ames-prod.hoon
│   ├── ames-sift.hoon
│   ├── ames-snub.hoon
│   ├── ames-verb.hoon
│   ├── ames-wake.hoon
│   ├── approve-merge.hoon
│   ├── autocommit.hoon
│   ├── automass.hoon
│   ├── bump.hoon
│   ├── cancel-autocommit.hoon
│   ├── cancel-automass.hoon
│   ├── cancel.hoon
│   ├── claz-invites.hoon
│   ├── close-flows.hoon
│   ├── code.hoon
│   ├── commit.hoon
│   ├── cors-approve.hoon
│   ├── cors-reject.hoon
│   ├── cp.hoon
│   ├── doff.hoon
│   ├── exit.hoon
│   ├── fuse
│   │   └── help.txt
│   ├── fuse.hoon
│   ├── fuse-list.hoon
│   ├── gall-sear.hoon
│   ├── gall-sift.hoon
│   ├── gall-verb.hoon
│   ├── global-automerge.hoon
│   ├── goad-gall.hoon
│   ├── hi.hoon
│   ├── init-oauth2
│   │   └── google.hoon
│   ├── init-oauth2.hoon
│   ├── install.hoon
│   ├── jump
│   │   ├── approve.hoon
│   │   ├── propose.hoon
│   │   └── reject.hoon
│   ├── keen.hoon
│   ├── knob.hoon
│   ├── label.hoon
│   ├── link.hoon
│   ├── mass.hoon
│   ├── meld.hoon
│   ├── merge
│   │   └── help.txt
│   ├── merge.hoon
│   ├── moon-breach.hoon
│   ├── moon-cycle-keys.hoon
│   ├── moon.hoon
│   ├── mount.hoon
│   ├── mv.hoon
│   ├── new-desk.hoon
│   ├── norm.hoon
│   ├── nuke.hoon
│   ├── ota.hoon
│   ├── pack.hoon
│   ├── pass.hoon
│   ├── pause.hoon
│   ├── pick.hoon
│   ├── ping.hoon
│   ├── private.hoon
│   ├── public.hoon
│   ├── rein.hoon
│   ├── rekey.hoon
│   ├── revive.hoon
│   ├── rm.hoon
│   ├── schedule.hoon
│   ├── serve.hoon
│   ├── start.hoon
│   ├── story-init.hoon
│   ├── story-remove.hoon
│   ├── story-write.hoon
│   ├── suspend.hoon
│   ├── sync-automerge.hoon
│   ├── sync.hoon
│   ├── syncs.hoon
│   ├── tomb.hoon
│   ├── trim.hoon
│   ├── uninstall.hoon
│   ├── unlink.hoon
│   ├── unmount.hoon
│   ├── unsync.hoon
│   ├── verb.hoon
│   ├── wham.hoon
│   └── yawn.hoon
├── ivory.hoon
├── jumps.hoon
├── key.hoon
├── keys.hoon
├── kick.hoon
├── ls.hoon
├── norms.hoon
├── roller
│   ├── assign.hoon
│   ├── commit.hoon
│   ├── config.hoon
│   ├── endpoint.hoon
│   ├── frequency.hoon
│   ├── goerli.hoon
│   ├── local.hoon
│   ├── quota.hoon
│   ├── refuel.hoon
│   ├── setkey.hoon
│   └── slice.hoon
├── show-vat.hoon
├── solid.hoon
├── spider
│   ├── kill.hoon
│   ├── poke.hoon
│   ├── start.hoon
│   └── stop.hoon
├── spider-tree.hoon
├── sponsor.hoon
├── stale-flows.hoon
├── story-list.hoon
├── story-read.hoon
├── timers.hoon
├── tree.hoon
├── trouble.hoon
├── tx.hoon
├── updates.hoon
├── vat.hoon
├── vats.hoon
├── verb.hoon
└── who.hoon
```

```
ted
├── aqua
│   ├── ames.hoon
│   ├── behn.hoon
│   ├── dill.hoon
│   ├── eyre-azimuth.hoon
│   └── eyre.hoon
├── azimuth
│   ├── load.hoon
│   ├── snap-logs.hoon
│   └── snap-state.hoon
├── build-cast.hoon
├── build-file.hoon
├── build-mark.hoon
├── build-nave.hoon
├── build-tube.hoon
├── claz
│   └── prep-command.hoon
├── code.hoon
├── diff.hoon
├── dns
│   ├── address.hoon
│   └── auto.hoon
├── eth
│   ├── get-timestamps.hoon
│   ├── get-tx-receipts.hoon
│   ├── read-contract.hoon
│   └── send-txs.hoon
├── eth-watcher.hoon
├── eval.hoon
├── example-fetch.hoon
├── first.hoon
├── hi.hoon
├── jam-all-desks.hoon
├── keen.hoon
├── keen-shut.hoon
├── khan-eval.hoon
├── ph
│   ├── add.hoon
│   ├── all.hoon
│   ├── boot-az.hoon
│   ├── boot-planet.hoon
│   ├── breach-hi-aqua.hoon
│   ├── breach-hi-cousin.hoon
│   ├── breach-hi.hoon
│   ├── breach-multiple.hoon
│   ├── breach-sudden.hoon
│   ├── breach-sync.hoon
│   ├── change-file.hoon
│   ├── child-sync.hoon
│   ├── child-update.hoon
│   ├── hi-az.hoon
│   ├── hi-comet-az.hoon
│   ├── hi-cousin-az.hoon
│   ├── hi.hoon
│   ├── hi-linnup-az-backward.hoon
│   ├── hi-linnup-az.hoon
│   ├── hi-marbud-az.hoon
│   ├── hi-nephew-az.hoon
│   ├── hi-uncle-az.hoon
│   ├── keen.hoon
│   ├── moon-az.hoon
│   ├── second-cousin-hi.hoon
│   ├── start-drivers.hoon
│   └── tend.hoon
├── read.hoon
├── roller
│   ├── nonce.hoon
│   └── send.hoon
├── runtime-version.hoon
├── test.hoon
├── time.hoon
├── tree.hoon
└── work.hoon
```

Glancing over this list, we have some categories:

1. Sysadmin:  Dojo/Hood/Helm, `+vats`, `|verb`, etc.
2. Development:  `ivory`, etc.
3. Azimuth + L2 Roller
4. Miscellaneous:  `story`, etc.

Anything stick out that is definitely out of date?

- `story` was an interesting project but doesn't really have a use case, for instance.
- `|new-desk` needs an update for at least `/mar/mime`.

CC @belisarius222 @midden-fabler @Fang- @joemfb @tinnus-napbus @jalehman

**Comment by sigilante** [comment]
Gathering notes here:  https://docs.google.com/spreadsheets/d/13jG4ZP8zonsENuCZRbZeL6g7F2JW_yVh3_a04Cuc2XA/edit#gid=0

Initial suggestions:

1. Propose some modest reorganization (like moving Azimuth stuff together).
2. Update everything with a `+help`-compatible header.
3. Make a simple `+help` and a more complex `+manual` with more details for certain generators and apps.
4. Remove `story`.
5. Remove `aqua` generators until Aqua is working again.
6. Merge `vats` and `show-vat`.
7. Move pill makers into `/gen/pill` (e.g. `+pill/brass`).

**Comment by sigilante** [comment]
The [table](https://docs.google.com/spreadsheets/d/13jG4ZP8zonsENuCZRbZeL6g7F2JW_yVh3_a04Cuc2XA/edit?gid=0#gid=0) has a comprehensive set of recommendations now and should be globally visible.

I propose moving or soft-linking/duplicating several generators. This will affect documentation and some usage patterns, but it's all in `/gen` right now so it shouldn't impact app or kernel behavior.

**Comment by sigilante** [comment]
Plan is to propose/implement %base cleanup changes and improve +help docs (which are in shambles) and regular docs (which are mostly complete) at the same time.

**Comment by sigilante** [comment]
Mostly resolved by #7017.

Have not looked at removing `%aqua` generators or merging `vats`.
